core/src/main/java/org/elasticsearch/index/query/SimpleQueryStringBuilder.java

```diff
@@ -201,8 +403,102 @@ public void doXContent(XContentBuilder builder, Params params) throws IOExceptio
         builder.endObject();
     }
 
+    /**
+     * Returns the name of the query parser to use when parsing
+     * {@link SimpleQueryStringBuilder} objects from JSON.
+     */
     @Override
     public String queryId() {
         return NAME;
     }
+
+    /**
+     * Reads a {@link SimpleQueryStringBuilder} object from the given input
+     * stream.
+     * 
+     * @param in Stream to read the object from.
+     * @return the object read from the stream.
+     * */
+    @Override
+    public SimpleQueryStringBuilder readFrom(StreamInput in) throws IOException {
+        String text = in.readOptionalString();
+        float boost = in.readFloat();
+        int size = in.readInt();
+        Map<String, Float> fields = new HashMap<>();
+        for (int i = 0; i < size; i++) {
+            String field = in.readString();
+            Float weight = in.readFloat();
+            fields.put(field, weight);
+        }
+
+        int flags = in.readInt();
+        String analyzer = in.readOptionalString();
+
+        Operator operator = null;
+        operator = Operator.parseFromInt(in.readInt());
+        Boolean lowercaseExpandedTerms = in.readBoolean();
+        Boolean lenient = in.readBoolean();
+        Boolean analyzeWildcards = in.readBoolean();
+
+        String localeStr = in.readString();
+        Locale locale = Locale.forLanguageTag(localeStr);
+
+        String queryName = (in.readOptionalString());
+        String minimumShouldMatch = in.readOptionalString();
+
+        SimpleQueryStringBuilder result = new SimpleQueryStringBuilder(text);
+        result.boost(boost).fields(fields).analyzer(analyzer).queryName(queryName).defaultOperator(operator);
+        result.minimumShouldMatch(minimumShouldMatch).locale(locale).lowercaseExpandedTerms(lowercaseExpandedTerms);
+        result.lenient(lenient).analyzeWildcard(analyzeWildcards).flags(flags);
```

Comment:
why do we save all the values to variables and then assign using setters? how about doing result.boost = in.readFloat() for instance and so on?